cli/src/cmd/cast/send.rs

```diff
@@ -138,29 +129,39 @@ impl SendTxArgs {
                 to_json,
             )
             .await
-        } else if config.sender != Config::DEFAULT_SENDER {
-            // Checking if signer isn't the default value
-            // 00a329c0648769A73afAc7F9381E08FB43dBEA72.
+        // Case 2:
+        // An option to use a local signer was provided.
+        // If we cannot successfully instanciate a local signer, then we will assume we don't have
+        // enough information to sign and we must bail.
+        } else {
+            // Retrieve the signer, and bail if it can't be constructed.
+            let signer = eth.wallet.signer(chain.id()).await?;
+            let from = signer.address();
+
+            // prevent misconfigured hwlib from sending a transaction that defies
+            // user-specified --from
+            if let Some(specified_from) = eth.wallet.from {
+                if specified_from != from {
+                    eyre::bail!(
+                        r#"
+The specified sender via CLI/env vars does not match the sender configured via
+the hardware wallet's HD Path.
+Please use the `--hd-path <PATH>` parameter to specify the BIP32 Path which
+corresponds to the sender, or let foundry automatically detect it by not specifying any sender address.
+"#
```

Comment:
Same for all the other strings. Either escape newlines or remove them

```rust
r#"
a
"# == "\na\n"

"\
a\
" == "a"

"\
a" == "a"
```

```suggestion
                        "\
The specified sender via CLI/env vars does not match the sender configured via
the hardware wallet's HD Path.
Please use the `--hd-path <PATH>` parameter to specify the BIP32 Path which
corresponds to the sender, or let foundry automatically detect it by not specifying any sender address.\
"
```

Reply:
Hey @DaniPopes this was done after a previous review by @0xMelkor — happy to accommodate whatever style you prefer, would just like alignment on which one we all prefer 😅  (and the current style is already used across the codebase)

Reply:
The message as is now will be printed with trailing newlines as I was trying to highlight in the first code block, and you don't need raw strings to allow newlines in literals: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=9652a8796a70c06b62941ce502bec872
I saw that each line ended with `"\n\"`, which is superfluous, but this suggestion is not a style choice

Reply:
Ahh I see what you mean now, my bad! Awesome will incorporate it then

Reply:
done in https://github.com/foundry-rs/foundry/pull/4874/commits/7da515ee9756a8c46b6656bbec910b25980e9a0b